apps/ensindexer/src/handlers/Resolver.ts

```diff
@@ -323,10 +325,72 @@ export const makeResolverHandlers = ({ pluginName }: { pluginName: PluginName })
         node: Node;
         name: Hex;
         resource: number;
+        ttl?: number;
         record: Hex;
       }>;
     }) {
-      // subgraph ignores
+      // subgraph ignores this event
+      if (pluginName === PluginName.Subgraph) return;
+
+      // but for non-subgraph plugins, we parse the RR set data for relevant records
+      const { node, name, resource, record } = event.args;
+
+      // we only index TXT records (resource id 16)
+      if (resource !== 16) return;
+
+      // parse the record's name, which is the key of the DNS record
+      const [, recordName] = decodeDNSPacketBytes(hexToBytes(name));
+
+      // invariant: recordName is always available and parsed correctly
+      if (!recordName) throw new Error(`Invalid DNSPacket, cannot parse name '${name}'.`);
+
+      // relevant keys end with .ens
+      if (!recordName.endsWith(".ens")) return;
+
+      // trim the .ens off the end to match ENS record naming
+      const key = recordName.slice(0, -4);
+
+      // parse the `record` parameter, which is an RRSet describing the value of the DNS record
+      const answers = parseRRSet(record);
+      for (const answer of answers) {
+        switch (answer.type) {
+          case "TXT": {
+            // > When decoding, the return value will always be an array of Buffer.
+            // https://github.com/mafintosh/dns-packet
+            const value = decodeTXTData(answer.data as Buffer[]);
+
+            const id = makeResolverId(event.log.address, node);
+            const resolver = await upsertResolver(context, {
+              id,
+              domainId: node,
+              address: event.log.address,
+            });
+
+            // upsert new key
+            await context.db
+              .update(schema.resolver, { id })
+              .set({ texts: uniq([...(resolver.texts ?? []), key]) });
+
+            // log ResolverEvent
+            await context.db.insert(schema.textChanged).values({
+              ...sharedEventValues(context.network.chainId, event),
+              resolverId: id,
+              key,
+              // note: coalesce empty string to null, see `handleTextChanged` for context
+              value: value || null,
+            });
+            break;
+          }
+          default: {
+            // no-op unhandled records
+            // NOTE: should never occur due to resource id check above
+            console.warn(
+              `Invariant: received answer ${JSON.stringify(answer)} that is not type === TXT despite resource === 16 check!`,
+            );
+            break;
+          }
+        }
+      }
```

Comment:
Making exceptions based on the plugin makes me think we could inject the new logic behind an interface from a particular plugin. Meaning, the `subgraph` plugin would implement a no-op for that interface, and both `basenames` and `lineanames` would implement the interface with the exact logic you proposed above.

Thoughts on that approach?

Reply:
now that we have the tooling to confirm that subgraph compatibility isn't affected by changes to the shared handlers, i'm less worried about including per-plugin logic here